common-kafka/src/main/java/com/cerner/common/kafka/consumer/ProcessingKafkaConsumer.java

```diff
@@ -194,6 +195,21 @@ public ProcessingKafkaConsumer(ProcessingConfig config) {
         this(config, new KafkaConsumer<>(config.getProperties()));
     }
 
+    /**
+     * Creates a new {@link ProcessingKafkaConsumer} with the given configuration and deserializers using the {@link KafkaConsumer}.
+     *
+     * @param config the configuration used by the consumer
+     * @param keyDeserializer The deserializer instance for key. When passing the deserializer instance
+     *          directly, the {@link Deserializer#configure(Map, boolean)} method will not be called.
+     * @param valueDeserializer The deserializer instance for value. When passing the deserializer instance
+     *          directly, the {@link Deserializer#configure(Map, boolean)} method will not be called.
+     * @throws NullPointerException if config is {@code null}
```

Comment:
would you mind fixing this here (should be IllegalArgumentException) and for the previously existing constructor?

A new static private method extracting the properties from the config could throw the IAE if it's null

Reply:
https://github.com/cerner/common-kafka/pull/8/commits/291376e1fb68c4fb2fd3ae0ed9cf68a338fda504